Core/Core/Assignments/AssignmentDetails/AssignmentDetailsView.swift

```diff
@@ -41,27 +43,30 @@ public struct AssignmentDetailsView: View, ScreenViewTrackable {
         screenViewTrackingParameters = ScreenViewTrackingParameters(
             eventName: "/courses/\(courseID)/assignments/\(assignmentID)"
         )
-
+        isTeacherEnrollment = course.first?.enrollments?.contains(where: { $0.isTeacher || $0.isTA }) == true
+        isLocked = assignment.first?.lockedForUser ?? false
     }
 
     public var body: some View {
         states
             .background(Color.backgroundLightest)
             .navigationBarStyle(.color(course.first?.color))
             .navigationTitle(NSLocalizedString("Assignment Details", comment: ""), subtitle: course.first?.name)
-            .rightBarButtonItems {
-                [
-                    UIBarButtonItemWithCompletion(
-                        title: NSLocalizedString("Edit", comment: ""),
-                        actionHandler: {
-                            env.router.route(
-                                to: "courses/\(courseID)/assignments/\(assignmentID)/edit",
-                                from: controller,
-                                options: .modal(isDismissable: false, embedInNav: true)
-                            )
-                        }
-                    ),
-                ]
+            .if(isTeacherEnrollment) { states in
```

Comment:
We marked this `if` view modifier as deprecated so let's not use it. Can we make a factory method for the `rightBarButtonItems` array and return an empty one if the enrollment is not a teacher one?

Reply:
refactored.